### PR TITLE
Upgrade bazel to 0.23.2 for docker legacy bazel builds

### PR DESCRIPTION
**Pull request by billfeng327**
After #18643 bazel RBE was updated to 0.23.2, this change updates the Dockerfile for legacy bazel builds (grpc_bazel_build Kokoro jobs) to use the latest version of bazel.

This also disables examples/... from being built by Kokoro C/C++ legacy CI jobs, as they include py_proto_library targets that cannot be filtered.

This PR temporarily disables grpc_python_bazel_test Kokoro CI job by removing the "bazel test" statement.

All excluded targets will be added back once the issue with python rules for bazel 0.23.0+ is resolved.

### CODE REVIEW

**Comment by jtattermusch** [comment]
A few remarks:
- if you are disabling python tests, have this PR approved @gnossen or at least talk to him and agree on an plan to add back (you can file a tracking issue for that if reintroducing python bazel test will take longer)
- just for clarification, the non-RBE bazel tests aren't "legacy" in the sense that they will be going away or that they are less legitimate than the RBE tests - they are simply tests that try to do a local bazel build and we will probably keep them around to make sure local bazel build continues working.

**Comment by jtattermusch** [comment]
FYI "Bazel Basic Tests for Python (Local)" can be found here:
https://github.com/grpc/grpc/blob/master/tools/internal_ci/linux/grpc_python_bazel_test_in_docker.sh (they are currently failing).

File: tools/internal_ci/linux/grpc_bazel_build_in_docker.sh [hunk 1]
```diff
@@ -24,4 +24,4 @@ git clone /var/local/jenkins/grpc /var/local/git/grpc
 && git submodule update --init --reference /var/local/jenkins/grpc/${name} \
 ${name}')
 cd /var/local/git/grpc
-bazel build --spawn_strategy=standalone --genrule_strategy=standalone :all test/... examples/...
+bazel build --spawn_strategy=standalone --genrule_strategy=standalone :all test/...
```
Comment (gnossen) [inline tools/internal_ci/linux/grpc_bazel_build_in_docker.sh]: I'm fine with removing these for now, but can we put a `TODO` here to add them back?

**Comment by billfeng327** [comment]
> FYI "Bazel Basic Tests for Python (Local)" can be found here:
> https://github.com/grpc/grpc/blob/master/tools/internal_ci/linux/grpc_python_bazel_test_in_docker.sh (they are currently failing).

Disabled the tests from that job

**Pull request merged**